airbyte-integrations/connectors/destination-bigquery-denormalized/src/test/java/io/airbyte/integrations/destination/bigquery/BigQueryDenormalizedDestinationTest.java

```diff
@@ -0,0 +1,424 @@
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldList;
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableDefinition;
+import io.airbyte.integrations.base.AirbyteStreamNameNamespacePair;
+import io.airbyte.integrations.destination.bigquery.formatter.BigQueryRecordFormatter;
+import io.airbyte.integrations.destination.bigquery.formatter.DefaultBigQueryDenormalizedRecordFormatter;
+import io.airbyte.integrations.destination.bigquery.formatter.GcsBigQueryDenormalizedRecordFormatter;
+import io.airbyte.integrations.destination.bigquery.formatter.arrayformater.LegacyArrayFormatter;
+import io.airbyte.integrations.destination.bigquery.uploader.AbstractBigQueryUploader;
+import io.airbyte.integrations.destination.bigquery.uploader.BigQueryDirectUploader;
+import io.airbyte.integrations.destination.bigquery.uploader.BigQueryUploaderFactory;
+import io.airbyte.integrations.destination.bigquery.uploader.UploaderType;
+import io.airbyte.integrations.destination.bigquery.uploader.config.UploaderConfig;
+import io.airbyte.protocol.models.AirbyteStream;
+import io.airbyte.protocol.models.ConfiguredAirbyteStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class BigQueryDenormalizedDestinationTest {
+
+  @Mock
+  UploaderConfig uploaderConfigMock;
+  @Mock
+  ConfiguredAirbyteStream configuredStreamMock;
+  @Mock
+  AirbyteStream airbyteStreamMock;
+  @Mock
+  DefaultBigQueryDenormalizedRecordFormatter bigQueryRecordFormatterMock;
+  @Mock
+  BigQuery bigQueryMock;
+
+  MockedStatic<BigQueryUploaderFactory> uploaderFactoryMock;
+
+  @InjectMocks
+  BigQueryDenormalizedDestination bqdd;
+
+  @BeforeEach
+  void init() {
+    uploaderFactoryMock = Mockito.mockStatic(BigQueryUploaderFactory.class, Mockito.CALLS_REAL_METHODS);
+    uploaderFactoryMock.when(() -> BigQueryUploaderFactory.getUploader(any(UploaderConfig.class))).thenReturn(mock(BigQueryDirectUploader.class));
+  }
+
+  @AfterEach
+  public void teardown() {
+    uploaderFactoryMock.close();
+  }
+
+  @Test
+  void getTargetTableName() {
+    final String streamName = "stream_name";
+    BigQuerySQLNameTransformer nameTransformerMock = mock(BigQuerySQLNameTransformer.class);
+    bqdd.getTargetTableName(streamName);
+    verify(nameTransformerMock, times(0)).getIdentifier(streamName);
```

Comment:
Good question. Probably I was thinking how should I test it but just forgot to edit it 😕. The method `getTargetTableName` in BigQueryDenormalizedDestination class does nothing but only calling namingResolver.getIdentifier(streamName) which is created in default constructor. 

I updated it to test the expected result but question for me is should we test functionality from others classes like `BigQuerySQLNameTransformer` to be able to fill requirement of 90 % coverage since in this case we test this functionality multiple times in a different Unit tests

Reply:
yeah it feels redundant to test both of those, and I'd prefer to remove this one...

I'm OK with having a single test case here just to satisfy the coverage counter, but would be nice to just not have this one. (I'm assuming that the actual NamingResolver has much more extensive testing, too)